Fix for triangular.pdf issue #101

akrawitz opened the pull request:
Fix issue #101 by updating the PDF for the triangular distribution, so that it returns the
proper value for an x that is less than, equal to, or greater than the
parameter c.

trevnorris commented:
Please run `npm test`. This PR causes a triangular pdf test to fail, and I'm unsure whether it's the test or the code that needs to be fixed.

Also, don't worry about getting commits looking just right. I can clean them up on my end. :)

akrawitz commented:
Tests have been updated.

`assert.epsilon(tol, jStat.triangular.pdf(1.0, 1.0, 2.0, 1.0), 1.0);`
This test was failing, however the test is incorrect. A comment in the
test code claimed that this test was based on R's `dtriangle(x, a, b, c)`
from package 'triangle', however this is not consistent with the current
return value for the R code:
`> dtriangle(1, 1, 2, 1)`
`[1] NaN`
`Warning message:`
`In dTest(c(x, a, b, c)) :`
`values required to be  a <= c <= b (at least one strict inequality)`

To further confuse matters, `dtriangle()` has a bug in it, given
that NaN is not the correct value for this case, and the provided
warning is not even consistent with the passed parameters! 

So I am now using R's `dtriang()` from package mc2d as my reference, as it
appears to be a better implementation of the triangular distribution.

I have added the reference R code into the comment in the test file,
since if we are going to refer to R functions for tests, we might as
well make it easy for someone to confirm.

Furthermore, I have added more tests, in an attempt to get basic
coverage of the various segments of the triangular distribution for each
of the basic forms it can take.

trevnorris commented:
Thanks. Landed in b6d945e, d0657dd and ea924b0.
